File: app/src/main/java/com/example/chenyong/android_demo/RxBus.java
package com.example.chenyong.android_demo;

import rx.Observable;
import rx.subjects.PublishSubject;
import rx.subjects.SerializedSubject;
import rx.subjects.Subject;

/**
 * Created by focus on 17/1/19.
 */

public class RxBus {
    private final Subject<Object, Object> _bus = new SerializedSubject<>(PublishSubject.create());

    public void send(Object o) {
        _bus.onNext(o);
    }

    public Observable<Object> toObserverable() {
        return _bus;
    }
}

File: app/src/main/kotlin/com/example/chenyong/android_demo/RxBus.kt
package com.example.chenyong.android_demo

import rx.Observable
import rx.subjects.PublishSubject
import rx.subjects.SerializedSubject
import rx.subjects.Subject

/**
 * Created by focus on 17/1/23.
 */
object RxBus {
    private val _bus: Subject<Any, Any> = SerializedSubject(PublishSubject.create())
    fun send(o: Any) {
        _bus.onNext(o)
    }

    fun toObserverable(): Observable<Any> {
        return _bus
    }

}